### ISSUE TITLE
Qt + (Open)SSL (on Android)

### ISSUE THREAD

**Issue by akikoskinen** [ruslo/hunter#1490]
* I've read [Brief overview section](https://docs.hunter.sh/en/latest/overview.html) and do understand basic concepts. **[Yes]**
* I've read [F.A.Q. section](https://docs.hunter.sh/en/latest/faq.html) and haven't found an answer to my question. **[Yes]**
* I've read [Code of Conduct](https://github.com/ruslo/hunter/blob/master/.github/CODE_OF_CONDUCT.md), I promise to be polite and will do my best at being constructive. **[Yes]**

---

Just today I learned about Hunter and played with it during the day. I tried whether it seems possible to handle package dependencies in an existing project. This project already uses CMake for building and dependencies include Qt. I also used Android as the target platform. Starting was easy enough.

I got Qt build, but then ran to a wall. The project uses QtNetwork and SSL is a requirement. On Android that means that Qt should be built with OpenSSL support (and maybe OpenSSL needs to be built before Qt?).

I'm not too familiar yet with how Hunter completely works, but I found [this line of code](https://github.com/ruslo/hunter/blob/fe957dec7c243577a336fb59bb79e499fba2a417/cmake/projects/Qt/schemes/url_sha1_qt.cmake.in#L266) that seems to disable SSL unconditionally. Am I interpreting this correctly?

If so, what would it need to obtain a Qt+OpenSSL+Android combo with Hunter? Would that need changes to Hunter? Somewhere else?

**Comment by ruslo** [comment]
> Would that need changes to Hunter?

Yes, since Qt use custom build scheme if you want to inject OpenSSL you have to add `hunter_add_package(OpenSSL)` and make sure Qt build can pick OpenSSL from Hunter. Probably something like this:
* https://github.com/ruslo/hunter/blob/fe957dec7c243577a336fb59bb79e499fba2a417/cmake/projects/Qt/schemes/url_sha1_qt.cmake.in#L186-L192

**Comment by xsacha** [comment]
I add these lines to all my Android Qt builds:

```
         -openssl-linked
         -I/path/to/android-ssl/include
         -L/path/to/android-ssl/lib
```

**Comment by akikoskinen** [comment]
Thanks so far.

@xsacha Is what you described a) something that you can do already with Hunter, b) something that you could do with Hunter, if something was slightly different in Hunter, or c) something you do when not using Hunter?

**Comment by xsacha** [comment]
B and C
